File: 3° semestre/Tecnicas-de-programacão-2/aula-24-09-06/estruturais/adapter/Adapter01.py
# interface do cliente
class Target:
    def request(self):
        pass

# cliente
class Cliente:
    def __init__(self, target):
        self.target = target

    def criarRequest(self):
        print("cliente fazendo uma requisição")
        self.target.request()

# serviço existente
class Adaptee:
    def resquestEspecifico(self):
        print("requisição especifica do adaptee")

# adaptador
class Adapter:
    def __init__(self,adaptee):
        self.adaptee = adaptee

    def request(self):
        self.adaptee.requestEspecifico()

# utilização do adaptador:
adaptee = Adaptee()
adapter = Adapter(adaptee)
cliente = Cliente(adapter)

cliente.criarRequest()

# cliente faz uma requisição e a requisição especifica do adaptee deve ser demonstrada

File: 3° semestre/Tecnicas-de-programacão-2/aula-24-09-06/estruturais/adapter/adapter01.js
// interface do cliente
class Target{
    resquest(){}
}

// cliente
class Cliente{
    constructor(target){
        console.log("cliente fazendo uma requisição")
        this.target.resquest();
    }
}

// serviço existente
class Adaptee{
    requestEspecifico(){
        console.log("Requsisição especifica do adaptee")
    }
}

// adaptador
class Adapter{
    constructor(Adaptee){
        this.adaptee = Adaptee;
    }

    request(){
        this.Adaptee.requestEspecifico();
    }
}

// utilização do adaptador
const adapte = new Adaptee();
const adapter = new Adapter(adaptee);
const cliente = new Cliente(adapter);

cliente.criarRequest();

// cliente faz uma requisição e a requisição especifica do adaptee deve ser demonstrada